baseplate/sidecars/live_data_watcher.py

```diff
@@ -115,17 +116,39 @@ def _parse_loader_type(data: bytes) -> LoaderType:
     return LoaderType(loader_type)
 
 
+def _generate_sharded_file_key(num_file_shards: Optional[int], file_key: str) -> str:
+    # We can't assume that every ZK Node that is being NodeWatched by the live-data-fetcher
+    # will make use of S3 prefix sharding - but, we know at least one does (/experiments).
+    # If it's not present or the value is 1, set the prefix to empty string ""
+    if not num_file_shards or num_file_shards == 1:
+        sharded_file_key_prefix = ""
+    else:
+        # If the num_file_shards key is present, we may have multiple copies of the same manifest
+        # uploaded so fetch one randomly using a randomly generated prefix.
+        # Generate a random number from 1 to num_file_shards exclusive to use as prefix.
+        sharded_file_key_prefix = str(random.randrange(1, num_file_shards)) + "/"
```

Comment:
I'm a little confused by `random.randrange()` -- it seems like it'll choose a number between 1 and `num_file_shards - 1`? I ran it a bunch in a repl and never got a 9, until I bumped the end range to 10:

```
>>> print(random.randrange(3, 9))
3
>>> print(random.randrange(3, 9))
3
>>> print(random.randrange(3, 9))
6
>>> print(random.randrange(3, 9))
8
>>> print(random.randrange(3, 9))
8
>>> print(random.randrange(3, 9))
4
>>> print(random.randrange(3, 9))
3
>>> print(random.randrange(3, 9))
4
>>> print(random.randrange(3, 9))
4
>>> print(random.randrange(3, 9))
6
>>> print(random.randrange(3, 9))
3
>>> print(random.randrange(3, 9))
5
>>> print(random.randrange(3, 9))
7
>>> print(random.randrange(3, 9))
7
>>> print(random.randrange(3, 9))
6
>>> print(random.randrange(3, 9))
7
>>> print(random.randrange(3, 9))
6
>>> print(random.randrange(3, 9))
8
>>> print(random.randrange(3, 9))
8
>>> print(random.randrange(3, 9))
6
>>> print(random.randrange(3, 9 + 1))
8
>>> print(random.randrange(3, 9 + 1))
9
>>> print(random.randrange(3, 9 + 1))
6
>>> print(random.randrange(3, 9 + 1))
8
>>> print(random.randrange(3, 9 + 1))
4
```

Do you want `num_file_shards + 1` here?

Reply:
Yup, it's meant to be exclusive. When NUM_FILE_SHARDS = 6 for example, here are the files generated

file.json
1_file.json
2_file.json
3_file.json
4_file.json
5_file.json

So we only ever want to go up to 5.